packages/puppeteer-core/src/api/ElementHandle.ts

```diff
@@ -276,12 +272,17 @@ export abstract class ElementHandle<
     pageFunction: Func | string,
     ...args: Params
   ): Promise<HandleFor<Awaited<ReturnType<Func>>>> {
+    pageFunction = withSourcePuppeteerURLIfNone(
```

Comment:
Why not add to this method?

Reply:
Because `withSourcePuppeteerURLIfNone` relies on the callstack which gets modified with decorators.